microcosm_pubsub/dispatcher.py

```diff
@@ -30,40 +31,60 @@ def handle_batch(self, bound_handlers) -> List[MessageHandlingResult]:
 
         """
         return [
-            MessageHandlingResult.invoke(
-                func=self.handle_message,
-                message=message,
-                bound_handlers=bound_handlers
-            )
+            self.handle_message(message, bound_handlers)
             for message in self.sqs_consumer.consume()
         ]
 
-    def handle_message(self, message, bound_handlers):
+    def handle_message(self, message, bound_handlers) -> MessageHandlingResult:
         """
-        Handle a single message.
+        Handle a message.
 
         """
-        message_id = message.message_id
-        media_type = message.media_type
+        # initialize a context for the current message
+        with self.opaque.initialize(self.sqs_message_context, message):
+            # save elapsed time information to the context
+            with elapsed_time(self.opaque):
+                message_handling_result = MessageHandlingResult.invoke(
+                    func=self._handle_message,
+                    message=message,
+                    bound_handlers=bound_handlers,
+                )
+            message_handling_result.elapsed_time = self.opaque["elapsed_time"]
```

Comment:
Notice how this nesting works. We set `elapsed_time` outside of the inner nesting but inside the outer one.